models.py

```diff
@@ -1,12 +1,118 @@
-from sqlalchemy import Column, Integer, String, Text #type: ignore
+import uuid
+from enum import IntEnum
+
+from sqlalchemy import (
+    Column,
+    Integer,
+    String,
+    Text,
+    ForeignKey,
+    DateTime,
+    UniqueConstraint,
+    Index,
+)
+from sqlalchemy.types import TypeDecorator
+from sqlalchemy.orm import relationship
 from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.inspection import inspect
+from sqlalchemy.dialects.postgresql import UUID
+
 
 Base = declarative_base()
 
-class UserToken(Base):
-    __tablename__ = 'user_tokens'
 
-    id = Column(Integer, primary_key=True, autoincrement=True)
+class RoleEnum(IntEnum):
+    sender = 1
+    recipient = 2
+
+
+class IntEnumType(TypeDecorator):
+    impl = Integer
+
+    def __init__(self, enumtype, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._enumtype = enumtype
+
+    def process_bind_param(self, value, dialect):
+        return int(value) if value is not None else None
+
+    def process_result_value(self, value, dialect):
+        return self._enumtype(value) if value is not None else None
+
+
+class Serializer:
```

Comment:
Along with Serializer check whether we can use pydantic to support mapping between SQLAlchemy objects and Pydantic dataclass objects

Reply:
https://github.com/SAGNICKMAJUMDAR/Filter-Email/pull/10#discussion_r2107877365